compute_tools/src/spec.rs

```diff
@@ -423,11 +423,33 @@ pub fn handle_grants(node: &ComputeNode, client: &mut Client) -> Result<()> {
         );
         db_client.simple_query(&alter_query)?;
 
-        // // Explicitly grant CREATE ON SCHEMA PUBLIC to the web_access user.
-        // // This is needed since postgres 15, where this privilege is removed by default.
-        // let grant_query: String = "GRANT CREATE ON SCHEMA public TO web_access".to_string();
-        // info!("grant query for db {} : {}", &db.name, &grant_query);
-        // db_client.simple_query(&grant_query)?;
+        // For PostgreSQL version 15
+        // Explicitly grant CREATE ON SCHEMA PUBLIC to the web_access user.
+        // This is needed because since postgres 15 this privilege is removed by default.
+        let grant_query = "DO $$\n\
+                BEGIN\n\
+                    IF EXISTS(\n\
+                        SELECT nspname\n\
+                        FROM pg_catalog.pg_namespace\n\
+                        WHERE nspname = 'public'\n\
+                    ) AND\n\
+                    version() LIKE 'PostgreSQL 15%'\n\
```

Comment:
Is there any better way to check postgres major version than this?

Reply:
There is also `SHOW server_version;`, which shows just a numeric version without any prefixes and postfixes. Although, it cannot be used in expressions directly, but probably it's possible to save the result into var in plgpsql

Reply:
> it cannot be used in expressions directly

`current_setting('server_version') like '14%'`

Reply:
`server_version` is one of the `GUC_REPORT` settings, which means that the server always sends them to the client after authentication. tokio-postgres makes the `GUC_REPORT` settings available with the [Connection::parameter](https://docs.rs/tokio-postgres/latest/tokio_postgres/struct.Connection.html#method.parameter) function. So I'd suggest something like this:

```
// FIXME: I'm playing fast and loose with the datatypes here, this comparison probably won't work as it is. 
if db_client.parameter("server_version") > "14" {
   ...
} else {
   ...
}

Reply:
> server_version is one of the GUC_REPORT settings, which means that the server always sends them to the client after authentication. tokio-postgres makes the GUC_REPORT settings available with the [Connection::parameter](https://docs.rs/tokio-postgres/latest/tokio_postgres/struct.Connection.html#method.parameter) function. So I'd suggest something like this:

looks cool, but in this code we use a synchronous postgres client, that doesn't provide this feature(

Reply:
I don't see much difference between 
```version() LIKE 'PostgreSQL 15%'```
and
```current_setting('server_version') like '14%'```

so I'd keep it as is

Reply:
`server_version` is intended to be machine-parseable, while `version()` is a free-form string.

Reply:
There's also `server_version_num`